test/uk/gov/hmrc/economiccrimelevyregistration/forms/AmlStartDateFormProviderSpec.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.economiccrimelevyregistration.forms
+
+import play.api.data.FormError
+import uk.gov.hmrc.economiccrimelevyregistration.forms.behaviours.DateBehaviours
+import uk.gov.hmrc.economiccrimelevyregistration.utils.EclTaxYear
+
+class AmlStartDateFormProviderSpec extends DateBehaviours {
+  val form = new AmlStartDateFormProvider()()
+
+  "value" should {
+    val fieldName   = "value"
+    val requiredKey = "amlStartDate.error.allRequired"
+
+    behave like dateField(
+      form,
+      fieldName,
+      datesBetween(EclTaxYear.currentFinancialYearStartDate, EclTaxYear.currentFinancialYearEndDate)
+    )
+
+    behave like mandatoryDateField(
```

Comment:
Think we discussed in `mandatoryDateField` we could add some extra tests for different combinations of missing data

Reply:
👍